app/sprinkles/account/src/Controller/AccountController.php

```diff
@@ -405,6 +428,36 @@ public function login(Request $request, Response $response, $args)
 
         $currentUser = $authenticator->attempt(($isEmail ? 'email' : 'user_name'), $userIdentifier, $data['password'], $data['rememberme']);
 
+        // Check if the enforced password update setting is configured.
```

Comment:
As @lcharette has noted, this is a bit lengthy. Would be worth factoring this out to some common method. Same could possible be said for regular password resets, but I'll need to check the source in those areas first to determine if there is a gain to be made.

Reply:
I added methods to `PasswordSecurity` to make these checks. I think it cleaned up the code a lot but if my implementation is not what you had in mind please let me know.